### ISSUE TITLE
Bitwarden Update Appears to be deleting all application files Windows 10

### ISSUE THREAD

**Issue by Adams880** [bitwarden/desktop#546]
## Describe the Bug

For the last 2 versions (installed using a 1.21.0 installer), every time I update Bitwarden it breaks all shortcuts and then the Program Files/Bitwarden folder has only "Uninstall Bitwarden" file in it anymore. All other files are gone. I have not been able to figure out if they move to somewhere else or are just gone entirely.

## Steps To Reproduce

1. Install a 1.21.0 Version of Bitwarden
2. Update to a new version (I have been doing it on close, not off the prompt when opening, not sure it makes a difference). 
3. Files and shortcuts are deleted and I cannot open Bitwarden or find it in the windows key search.

## Expected Result

Bitwarden updates and does not uninstall itself on update. Also, shortcuts do not get broken when update is finished

## Actual Result

Bitwarden basically deletes itself except for the uninstall script

## Environment

- Operating system: Windows 10
- Build Version 1.21.0

**Comment by daniellutz** [comment]
This happened to me as well with the last versions of Bitwarden. =(

**Comment by clayadams5226** [comment]
@Adams880 this this happen to you with the most recent update? I was able to update here without any issue

**Comment by jacobvenborg** [comment]
I just updated and now it uninstalled itself.. Again. It happens to me very regularily.. Let me know if I can provide some information. I'm running Windows 10, version 10.0.1904, build 19042. Not sure which Bitwarden version I ran prior to upgrading. Probably the latest, since I do update regularily despite frequent uninstalls :)

**Comment by Adams880** [comment]
@clayadams5226 I just updated and it was no longer happening. I'm not really sure what the issue was.

**Comment by tallpants** [comment]
Just happened to me again, happy to provide any information that might help you replicate!

**Comment by jacobvenborg** [comment]
Yup, just happened here again. Likewise, I will gladly provide any information needed.

**Comment by tallpants** [comment]
@clayadams5226 I managed to record this happening on the newest update:

https://user-images.githubusercontent.com/15325890/111738191-e9990380-88a6-11eb-9c94-01d6fd68f470.mp4

And here's the logs from when this happened in case it helped:

```
[2021-03-19 11:23:44.087] [info] Checking for update
[2021-03-19 11:23:45.723] [info] Found version 1.25.1 (url: Bitwarden-Installer-1.25.1.exe)
[2021-03-19 11:23:45.724] [info] Downloading update from Bitwarden-Installer-1.25.1.exe
[2021-03-19 11:23:45.735] [info] Cached update sha512 checksum doesn't match the latest available update. New update must be downloaded. Cached: JwT/rU3R1zf07BF/yYPGRGQQYcqB8qS4rxfuXV9tmd8OObmKBsjuBGw4As+pa5m2sEFJ2YY+VRe5KQyfQ+c7uQ==, expected: eITD34oDPq5Crtvg9seFKmS5NWWsxabhs+K6grnLGdbTD8OCkc+HkDE3ZDzEncwLnWs2te0/y0AUPjS9U0Cbrg==. Directory for cached update will be cleaned
[2021-03-19 11:23:46.376] [error] Cannot download differentially, fallback to full download: Error: EPERM: operation not permitted, open 'C:\Users\Adithya Reddy\AppData\Local\bitwarden-updater\package.7z'
[2021-03-19 11:23:48.145] [info] New version 1.25.1 has been downloaded to C:\Users\Adithya Reddy\AppData\Local\bitwarden-updater\pending\Bitwarden-Installer-1.25.1.exe
[2021-03-19 11:28:18.525] [info] Install on explicit quitAndInstall
[2021-03-19 11:28:18.527] [info] Install: isSilent: false, isForceRunAfter: true
[2021-03-19 11:28:18.803] [info] Cannot run installer: error code: EACCES, error message: "spawn C:\Users\Adithya Reddy\AppData\Local\bitwarden-updater\pending\Bitwarden-Installer-1.25.1.exe EACCES", will be executed again using elevate if EACCES"
[2021-03-19 11:28:18.871] [info] Update installer has already been triggered. Quitting application.
```

Entire file: [app.log](https://github.com/bitwarden/desktop/files/6168925/app.log)

This is definitely 100% reproducible every single time on my computer and it's getting very very annoying.

**Comment by jvenborg** [comment]
The exact same thing just happened to me when trying to update to version 1.25.1 Still running Windows 10 Pro, 20H2, build 19042.867. I am not sure how to make logs, sorry, but will happily do it once I find out.

**Comment by szetyi** [comment]
Same thing happening to me too, I'm suspecting an error related to user folder with spaces is the cause, maybe special charachters, as @tallpants and I both have a space in our username

log:
`[2021-06-01 07:17:44.931] [info] Checking for update
[2021-06-01 07:17:46.281] [info] Found version 1.26.5 (url: Bitwarden-Installer-1.26.5.exe)
[2021-06-01 07:17:46.282] [info] Downloading update from Bitwarden-Installer-1.26.5.exe
[2021-06-01 07:17:46.302] [info] Update has already been downloaded to C:\Users\Szilágyi Péter\AppData\Local\bitwarden-updater\pending\Bitwarden-Installer-1.26.5.exe).
[2021-06-01 07:20:46.484] [info] Install on explicit quitAndInstall
[2021-06-01 07:20:46.485] [info] Install: isSilent: false, isForceRunAfter: true
[2021-06-01 07:20:46.744] [info] Cannot run installer: error code: EACCES, error message: "spawn C:\Users\Szilágyi Péter\AppData\Local\bitwarden-updater\pending\Bitwarden-Installer-1.26.5.exe EACCES", will be executed again using elevate if EACCES"
[2021-06-01 07:20:46.816] [info] Update installer has already been triggered. Quitting application.`

**Comment by jvenborg** [comment]
Sorry to bump, but it is still very much happening.

And what szetyi wrote is very interesting, I too have a space in my user name.

**Comment by Sc0tTyXL** [comment]
I've been having the same issue as well on two machines for months now:
Windows 10 20H2 on my laptop
Windows 10 20H1 on my desktop

And it stopped me te use the auto updater....

PS: Both machines have spaces in the username folder.

**Comment by MiniBossDK** [comment]
The error also happends for me.
Windows 10 20H1 and with spaces in the username folder.

**Comment by CarlBluePrism** [comment]
This also just happened to me, happy to provide any more details if needed.

**Comment by mjlehrke** [comment]
No spaces in my username and it removed itself.

**Comment by evanj97** [comment]
This just happened to me for the first time too.  
After reading this thread and 5-10 mins of googling, it looks like the error that szetyi provided: `EACCES` is an npm install error due to insufficient permissions to write to a directory (see below stack overflow links).  
I'm not familiar with how bitwarden works or installs and I've barely ever used node or JS/TS, but I would guess that npm is running from or accessing something in `AppData\Local\bitwarden-updater\pending\` (see szetyi's log) or some other temp folder instead of only touching the application directory.

Sources:
 [How to fix EACCES issues with npm install](https://stackoverflow.com/questions/25290986/how-to-fix-eacces-issues-with-npm-install) and [gyp WARN EACCES user "root" does not have permission to access the dev dir](https://stackoverflow.com/questions/29468404/gyp-warn-eacces-user-root-does-not-have-permission-to-access-the-dev-dir)
